src/node_api.cc

```diff
@@ -1024,6 +1024,27 @@ napi_status napi_delete_property(napi_env env,
   return GET_RETURN_STATUS(env);
 }
 
+NAPI_EXTERN napi_status napi_has_own_property(napi_env env,
+                                              napi_value object,
+                                              napi_value key,
+                                              bool* result) {
+  NAPI_PREAMBLE(env);
+  CHECK_ARG(env, key);
+
+  v8::Isolate* isolate = env->isolate;
+  v8::Local<v8::Context> context = isolate->GetCurrentContext();
+  v8::Local<v8::Object> obj;
+
+  CHECK_TO_OBJECT(env, context, obj, object);
+  v8::Local<v8::Value> k = v8impl::V8LocalValueFromJsValue(key);
+  RETURN_STATUS_IF_FALSE(env, k->IsName(), napi_name_expected);
```

Comment:
I don't think we expose it by itself. According to the spec, the key is converted to a name, which calls `toPrimitive`, [toPropertyKey](https://www.ecma-international.org/ecma-262/7.0/#sec-topropertykey). 

So, I think we should delete the `return false if not IsName()`, and let V8 handle it in a spec compliant way. V8 only needs a `value`, not a string/symbol here.